azurerm/resource_arm_managed_cluster.go

```diff
@@ -0,0 +1,556 @@
+package azurerm
+
+import (
+	"bytes"
+	"fmt"
+	"log"
+	"time"
+
+	"github.com/Azure/azure-sdk-for-go/services/containerservice/mgmt/2017-09-30/containerservice"
+	"github.com/hashicorp/terraform/helper/hashcode"
+	"github.com/hashicorp/terraform/helper/resource"
+	"github.com/hashicorp/terraform/helper/schema"
+	"github.com/hashicorp/terraform/helper/validation"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/utils"
+)
+
+func resourceArmManagedCluster() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceArmManagedClusterCreate,
+		Read:   resourceArmManagedClusterRead,
+		Update: resourceArmManagedClusterCreate,
+		Delete: resourceArmManagedClusterDelete,
+
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+
+			"location": locationSchema(),
+
+			"resource_group_name": resourceGroupNameSchema(),
+
+			"dns_prefix": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+
+			"kubernetes_version": {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+
+			"linux_profile": {
+				Type:     schema.TypeSet,
+				Required: true,
+				MaxItems: 1,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"admin_username": {
+							Type:     schema.TypeString,
+							Required: true,
+						},
+						"ssh_key": {
+							Type:     schema.TypeSet,
```

Comment:
if there's only a single item - we could probably just make this an array, what do you think? e.g.

```
ssh_key = ["${file(~/id_rsa.pub)"]
```

rather than

```
ssh_key {
  key_data = "${file(~/id_rsa.pub)"
}
```